File: blog/src/main/kotlin/com/liux/blog/bean/po/ArticleUrl.kt
package com.liux.blog.bean.po

import com.liux.blog.bean.NoArgsConstructor
import java.util.*

// 文章URL数据
@NoArgsConstructor
data class ArticleUrl(
    // URL
    var url: String? = null,
    // 文章ID
    var articleId: Int? = null,
    // URL状态 0_历史的 1_当前的
    var status: Int? = null,
    // 创建时间
    var createTime: Date? = null,
)

File: blog/service/article_url.go
package service

import (
	"blog/bean/po"
	"blog/database"

	"gorm.io/gorm"
	"gorm.io/gorm/clause"
)

type ArticleUrlService struct {
	Service
	db *gorm.DB
}

func (s *ArticleUrlService) OnInitService() {
	s.db = database.GetDB()
}

func (s *ArticleUrlService) QueryArticleUrlByUrl(url string) *po.ArticleUrl {
	var articleUrl po.ArticleUrl
	s.db.Model(&po.ArticleUrl{}).
		Where("? = ?", clause.Column{Name: "url"}, url).
		Take(&articleUrl)
	if articleUrl.Url == "" {
		return nil
	}
	return &articleUrl
}

func (s *ArticleUrlService) QueryArticleUrlById(id int) *po.ArticleUrl {
	var articleUrl po.ArticleUrl
	s.db.Model(&po.ArticleUrl{}).
		Where("? = ? AND ? = ?", clause.Column{Name: "status"}, po.ARTICLE_URL_STATUS_LAST, clause.Column{Name: "article_id"}, id).
		Take(&articleUrl)
	if articleUrl.Url == "" {
		return nil
	}
	return &articleUrl
}
